src/wutil.h

```diff
@@ -145,10 +145,15 @@ struct file_id_t {
     dev_t device{static_cast<dev_t>(-1LL)};
     ino_t inode{static_cast<ino_t>(-1LL)};
     uint64_t size{static_cast<uint64_t>(-1LL)};
-    time_t change_seconds{-1};
-    long change_nanoseconds{-1};
-    time_t mod_seconds{-1};
-    long mod_nanoseconds{-1};
+    /* some platforms handle negative time_t
+       values to represent WW1-era dates, initialize ancient
+       for the sake of comparisons.
+       tv_nsec's meaningful values are REALLY [0, 999999999]
+       this nanosecond component we'll initialize at 0. */
+    time_t change_seconds{std::numeric_limits<time_t>::min()};
+    long change_nanoseconds{0};
```

Comment:
sorry what I meant is that

- if we use `long`, then `std::numeric_limits<long>::min()` is the obvious "oldest" value
- if we use `unsigned long`, then 0 would be more readable than `std::numeric_limits<unsigned long>::min()`

`long 0` is a weird in-between state (how come it's better than the original -1).
I guess it makes no practical difference, but it would improve the clarity of the change to make it the actual oldest file that we can represent.

Reply:
I'm using std::numeric_limits<time_t>::min() for the `time_t` - that's a valid "really old" timestamp.
The nsec part is how many additional nanoseconds there are - it's defined between 0 and 1000000. Does INT_MIN make sense there?

Reply:
> how many additional nanoseconds there are - it's defined between 0 and 1000000

I see, then I'd stick to -1 since that's already older than any valid time. 0 is fine too IMO

Reply:
Makes sense.

Reply:
Those are left at -1 again now.